sorting_algorithms/cocktail_shaker_sort.pl

```diff
@@ -0,0 +1,55 @@
+% swap_up(+List, -ListWithMovedElement)
+% goes through the list
+% checks every two adjacent numbers
+% if the first is greater, then swap them
+% in other words, the biggest element is moved to position where after its there are only greater numbers.
+swap_up([], []).
+swap_up([X], [X]).
+swap_up([Fst, Snd|T],[Snd|L]) :- Fst > Snd, swap_up([Fst|T], L), !.
+swap_up([Fst, Snd|T],[Fst|L]) :- Fst =< Snd, swap_up([Snd|T], L).
```

Comment:
```suggestion
swap_up([Fst, Snd|T],[Fst|L]) :- swap_up([Snd|T], L).
```
Sorry, zapomniałem przy ostatnim:
jak masz odcięcie w poprzedniej lini to tu nie musisz sprawdzać Fst =< Snd